src/Eshopworld.Messaging/QueueAdapter.cs

```diff
@@ -68,9 +71,21 @@ internal async Task Send([NotNull]T message)
                 Label = message.GetType().FullName
             };
 
-            await Sender.SendAsync(qMessage).ConfigureAwait(false);
+            await SendPolicy.ExecuteAsync(async () =>
```

Comment:
Similar code is used in 7 places. Maybe it could be simplified?

Reply:
I think this should be abstracted to a Func.

Reply:
There are 4 variants to this `try catch` block:
- void for receivers
- void for senders
- with return type of receivers
- with return type for senders

@arturek @dguerin Can you specify if you are recommending 4 different methods to abstract 7 try catch blocks? Or are you recommending 2 different methods with catch block delegates to abstract the same 7 blocks?